From 6b4209b7f60c8ecde70cce9897ceb0815311b723 Mon Sep 17 00:00:00 2001
From: Adrian Alonso <adrian.alonso@nxp.com>
Date: Thu, 15 Sep 2016 15:49:41 -0500
Subject: [PATCH 1160/5242] MLK-13240: arm: imx6q: lpddr2 freq fix switch to
 100Mhz

commit  0e27911405f1b9f21022f3d4e37c1287442f77cc from
https://source.codeaurora.org/external/imx/linux-imx.git

Fix switch_to_100Mhz miss to store updated podf dividers
for system clocks running at 100Mhz (audio mode)

Signed-off-by: Adrian Alonso <adrian.alonso@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/lpddr2_freq_imx6q.S |    1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-imx/lpddr2_freq_imx6q.S b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
index d0ccb98..d032742 100644
--- a/arch/arm/mach-imx/lpddr2_freq_imx6q.S
+++ b/arch/arm/mach-imx/lpddr2_freq_imx6q.S
@@ -68,6 +68,7 @@ skip_periph_clk_switch_400m:
 	bic	r9, r9, r6
 	orr	r9, r9, #(0x1F << 16)
 	orr	r9, r9, #(0x1D << 8)
+	str	r9, [r2, #CCM_CBCDR]
 
 	wait_for_ccm_handshake
 
-- 
1.7.9.5

